numpy/core/arrayprint.py

```diff
@@ -650,18 +646,22 @@ def fillFormat(self, data):
                 format = '%'
             format = format + '%d.%de' % (self.max_str_len, self.precision)
         else:
-            format = '%%.%df' % (self.precision,)
             if len(non_zero):
-                precision = max([_digits(x, self.precision, format)
-                                 for x in non_zero])
+                precision = max([_trailing_nonzero(x, self.precision)
```

Comment:
yeah I worried about that too, but then I worked out that `data` is going to be a small array.

`data` only contains the elements that are actually printed on the screen (ie, not the interior which is replaced by `...`)